### PR TITLE
E2e revoke permissions

### PR DESCRIPTION
**Pull request by tmashuang**
## **Description**

E2e test for new wallet_revokePermissions (current unmerged https://github.com/MetaMask/metamask-extension/pull/21948). Once that PR is merged the base branch will need to target `develop`, this PR will be taken out of draft, and ready for review.

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/1706

## **Manual testing steps**

1. Pull down this branch
2. Run `yarn && yarn build:test && yarn test:e2e:chrome:rpc`
3. Should pass
4. Optional run `yarn test:e2e:single test/e2e/json-rpc/wallet_revokePermissions.spec.js --browser=chrome` should pass

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/json-rpc/wallet_revokePermissions.spec.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+        const result = await driver.executeScript(
+          `return window.ethereum.request(${revokePermissionsRequest})`,
+        );
+
+        // Response of method call
+        assert.deepEqual(result, null);
+
+        // Reload Dapp
+        await driver.executeScript(`window.location.reload()`);
```
Comment (tmashuang) [inline test/e2e/json-rpc/wallet_revokePermissions.spec.js:53]: Had to force reload the dapp for the accounts label to reset, but the normal behavior is that the label automatically clears with the permission revoked.

**Comment by HowardBraham** [comment]
How much work is it to convert this to TypeScript?  Sample TS spec files are in `/test/e2e/accounts/`

**Comment by tmashuang** [comment]
@HowardBraham Is this a requirement for getting this method tested?

**Comment by HowardBraham** [comment]
> @HowardBraham Is this a requirement for getting this method tested?

Not a requirement, just a suggestion, if it's not too hard.  We're supposed to be implementing everything new in TypeScript.

File: test/e2e/json-rpc/wallet_revokePermissions.spec.js (hunk 1 above)
Comment (adonesky1) [reply test/e2e/json-rpc/wallet_revokePermissions.spec.js:53]: > but the normal behavior is that the label automatically clears with the permission revoked.

are you saying its not auto clearing in the test but does so when manually testing?

**Review by adonesky1 — DISMISSED** [review]
LGTM! I guess it would be good to clear up the need to reload the dapp but not strictly blocking!

File: test/e2e/json-rpc/wallet_revokePermissions.spec.js (hunk 1 above)
Comment (jiexi) [reply test/e2e/json-rpc/wallet_revokePermissions.spec.js:53]: is this a test dapp bug then?

**Comment by jiexi** [comment]
What Alex said. I wouldn't mind just leaving a comment why reloading the dapp is necessary / when it would be okay to be removed

**Pull request closed without merging**